src/marshaller/HipieDDL.js

```diff
@@ -88,6 +53,54 @@
         return "string";
     }
 
+    //  Mappings ---
+    function SourceMappings(visualization, mappings) {
+        this.visualization = visualization;
+        var newMappings = {};
+        for (var key in mappings) {
+            if (mappings[key] instanceof Array) {
+                mappings[key].forEach(function (mapingItem, idx) {
+                    newMappings[idx === 0 ? key : key + "_" + idx] = mapingItem;
+                });
+            } else {
+                newMappings[key] = mappings[key];
+            }
+        }
+        this.mappings = newMappings;
+        this.hasMappings = false;
+        this.reverseMappings = {};
+        this.columns = [];
+        this.columnsIdx = {};
+        this.columnsRHS = [];
+        this.columnsRHSIdx = {};
+    }
+
+    SourceMappings.prototype.init = function() {
+        for (var key in this.mappings) {
+            this.reverseMappings[this.mappings[key]] = key;
+            if (this.columnsIdx[key] === undefined) {
+                this.columns.push(key);
+                this.columnsIdx[key] = this.columns.length - 1;
+            }
+            this.columnsRHS[this.columnsIdx[key]] = this.mappings[key];
+            this.columnsRHSIdx[this.mappings[key]] = this.columnsIdx[key];
+            this.hasMappings = true;
+        }
+    };
+
+    SourceMappings.prototype.init = function() {
```

Comment:
I think I just moved "hipieType2DBType" to the top of the file and this is how GitHub diff is showing it...